addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,216 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
     public void importCollection(String collection) throws JsonProcessingException {
         PostmanCollection postmanCollection = parse(collection);
 
-        // TODO: Extract list of HttpMessage from PostmanCollection and send requests
+        List<HttpMessage> httpMessages = new ArrayList<>();
+        extractHttpMessages(postmanCollection.getItem(), httpMessages);
+
+        requestor.run(httpMessages);
     }
 
     public PostmanCollection parse(String collectionJson) throws JsonProcessingException {
         ObjectMapper objectMapper = new ObjectMapper();
         return objectMapper.readValue(collectionJson, PostmanCollection.class);
     }
 
+    private void extractHttpMessages(List<AbstractItem> items, List<HttpMessage> httpMessages) {
+        for (AbstractItem item : items) {
+            if (item instanceof Item) {
+                HttpMessage httpMessage = extractHttpMessage((Item) item);
+                if (httpMessage != null) {
+                    httpMessages.add(httpMessage);
+                }
+            } else if (item instanceof ItemGroup) {
+                extractHttpMessages(((ItemGroup) item).getItem(), httpMessages);
+            }
+        }
+    }
+
     private static boolean isSupportedScheme(String scheme) {
         return "http".equalsIgnoreCase(scheme) || "https".equalsIgnoreCase(scheme);
     }
+
+    public static boolean isContentTypeAlreadySet(List<KeyValueData> headers) {
+        for (KeyValueData header : headers) {
+            if (header.getKey().equalsIgnoreCase("Content-Type")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static HttpMessage extractHttpMessage(Item item) {
+        Request request = item.getRequest();
+        Url url = request.getUrl();
+
+        if (request == null || url == null) {
+            return null;
+        }
+
+        HttpMessage httpMessage;
+        try {
+            String rawUrl = url.getRaw();
+            httpMessage = new HttpMessage(new URI(rawUrl, false));
+        } catch (URIException | HttpMalformedHeaderException | NullPointerException e) {
+            return null;
+        }
+
+        httpMessage.getRequestHeader().setMethod(request.getMethod());
+
+        for (KeyValueData header : request.getHeader()) {
+            if (!header.isDisabled()) {
+                httpMessage.getRequestHeader().setHeader(header.getKey(), header.getValue());
+            }
+        }
+
+        Body body = request.getBody();
+
+        if (body.isDisabled()) {
+            return httpMessage;
+        }
+
+        String mode = body.getMode();
+
+        String bodyContent = "";
+        String contentType = "";
+
+        if (mode == null || body == null) {
+            return httpMessage;
+        }
+        if (mode.equals("raw")) {
+            Map<String, String> contentTypeMap =
```

Comment:
Extract a constant. Is this documented somewhere? The schema does not seem to have these.

Reply:
Found [this](https://github.com/postmanlabs/schemas/blob/da7578c2d71c46de2d39d04fbeebc26570591a44/schemas/draft-07/v2.1.0/collection/request.json#L72) link from the doc

Reply:
But that does not document what's being done here.

Reply:
Same here :)

Reply:
As discussed in the weekly call, Postman automatically attaches the `Content-type` header (which is documented [here](https://learning.postman.com/docs/sending-requests/requests/#sending-body-data), therefore I am resolving this comment.

Reply:
Should still extract a constant.

Reply:
I think I didn't get it. Constant for what?

Reply:
From `contentTypeMap`, that's effectively constant.